src/native/minipal/memory.c

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "memory.h"
+
+#ifdef HOST_WINDOWS
+#include <Windows.h>
+
+void* minipal_co_task_mem_alloc(size_t cb)
+{
+    return CoTaskMemAlloc(cb);
+}
+
+void minipal_co_task_mem_free(void* pv)
+{
+    CoTaskMemFree(pv);
+}
+#else
+// CoTaskMemAlloc always aligns on an 8-byte boundary.
+#define ALIGN 8
+
+void* minipal_co_task_mem_alloc(size_t cb)
+{
+    // Ensure malloc always allocates.
+    if (cb == 0)
+        cb = ALIGN;
+
+    // Align the allocation size.
+    size_t cb_safe = (cb + (ALIGN - 1)) & ~(ALIGN - 1);
+    if (cb_safe < cb) // Overflow
+        return NULL;
+
+    return aligned_alloc(ALIGN, cb_safe);
```

Comment:
Maybe we can move this here (with cmake checks):

https://github.com/dotnet/runtime/blob/e377401c233637e761d597115ab42b1e23b757b4/src/native/libs/System.Native/pal_memory.c#L26